### PR TITLE
Do not redirect insecure requests

### PR DESCRIPTION
**Pull request by calebmer**
I was recently doing research and came across [this](https://github.com/interagent/http-api-design#require-secure-connections). I'll quote the relevant sections here:

> ### Require Secure Connections
>
> Require secure connections with TLS to access the API, without exception. It’s not worth trying to figure out or explain when it is OK to use TLS and when it’s not. Just require TLS for everything.
>
> Ideally, simply reject any non-TLS requests by not responding to requests for http or port 80 to avoid any insecure data exchange. In environments where this is not possible, respond with `403 Forbidden`.
>
> **Redirects are discouraged** since they allow sloppy/bad client behaviour without providing any clear gain. Clients that rely on redirects double up on server traffic and render TLS useless since sensitive data will already have been exposed during the first call.

Let's discuss potentially not redirecting (as is recommended in the last paragraph) and just responding with a 403. I like the idea 😊

### CODE REVIEW

**Comment by begriffs** [comment]
This makes a lot of sense.

File: src/PostgREST/Middleware.hs [hunk 1]
```diff
@@ -54,26 +50,14 @@ runWithClaims conf app req = do
...
+checkInsecure app req respond =
   if not (isSecure req || isHerokuSecure)
-    then case uriM of
-      Just uri ->
-        respond $ responseLBS status301 [
-            (hLocation, cs . show $ uri { uriScheme = "https:" })
-          ] ""
-      Nothing ->
-        respond $ responseLBS status400 [] "SSL is required"
+    then respond $ responseLBS status403 [] "SSL is required"
```
Comment (begriffs) [inline src/PostgREST/Middleware.hs]: Maybe a json object matching the other postgrest error messages, like

```json
{ "message": "SSL is required" }
```

Comment (calebmer) [inline src/PostgREST/Middleware.hs]: Oh, that doesn't automatically make an object? That won't be a problem.

Comment (begriffs) [inline src/PostgREST/Middleware.hs]: Yeah `responseLBS` just passes a string along as the response, no matter what the status code. You'll need to hard code the json in there and probably set the content type response header to say it's json.

**Comment by ruslantalpa** [comment]
One question, since postgrest does not support ssl (for that it needs nginx) when will the headers ever contain "https"?

**Comment by begriffs** [comment]
Once upon a time it used to support SSL and the secure redirection stuff comes from that time. Some people say we should make it support SSL once again.

**Comment by ruslantalpa** [comment]
"some say ..." :). I don't see why add the complexity and another dependency (ssl lib) when it can very well be handled by nginx. This code and flag can be removed all together (i would go as far as remove gzip and let that be also handled by nginx).

**Comment by begriffs** [comment]
Less code's the best code, what do you think @calebmer?

**Comment by calebmer** [comment]
I'm game for deleting SSL support entirely. Either we support SSL fully or not at all.

**Comment by calebmer** [comment]
Ok, just removed the secure flag entirely.

**Comment by begriffs** [comment]
How nice, looks cleaner.

**Pull request merged**